app/scripts/views/header.coffee

```diff
@@ -126,9 +128,17 @@ module.exports = class HeaderView extends View
     e.preventDefault()
     ViewHelpers.toggleAnonymousStatus()
 
+  SetRefreshAnonymousStatusInterval: ->
+    # Check anonymous status every 500ms and indicate it with icon
+    if ViewHelpers.isLogged() and not @RefreshAnonymousStatusInterval
+      @RefreshAnonymousStatusInterval = setInterval @RefreshAnonymousStatus, 500
```

Comment:
It's necessary for proper feedback. You're clicking on the button and expecting it will change status quickly. Changing button status can be implemented directly, but it will be duplicating code in interval loop which is necessary too.